pkg/controllermanager/controller/event/event_control.go

```diff
@@ -0,0 +1,93 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package event
+
+import (
+	"context"
+	"fmt"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap/keys"
+	"github.com/gardener/gardener/pkg/logger"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"time"
+)
+
+var nowFunc = time.Now
+
+var longLivedEventReasons = sets.NewString(
+	gardencorev1beta1.ShootEventImageUpdate,
+	gardencorev1beta1.ShootEventK8sVersionUpdate,
+	gardencorev1beta1.ShootEventMaintenanceError,
+	gardencorev1beta1.ShootEventHibernationEnabled,
+	gardencorev1beta1.ShootEventHibernationDisabled,
+)
+
+func (c *Controller) enqueueEvent(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		logger.Logger.Errorf("Couldn't get key for object %+v: %v", obj, err)
+		return
+	}
+	c.eventQueue.Add(key)
+}
+
+func (c *Controller) reconcileEvent(req reconcile.Request) (reconcile.Result, error) {
+	ctx := context.TODO()
+
+	gardenClient, err := c.clientMap.GetClient(ctx, keys.ForGarden())
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get garden client: %w", err)
+	}
+
+	log := logger.NewFieldLogger(logger.Logger, "event", req.Name)
+
+	event := &corev1.Event{}
+	if err := gardenClient.Client().Get(ctx, req.NamespacedName, event); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Debug("Skipping because Event has been deleted")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	if longLivedEvent(event) {
+		return reconcile.Result{}, nil
+	}
+
+	deleteAt := event.FirstTimestamp.Add(c.cfg.Controllers.Event.TTLShortLivedEvents.Duration)
+	timeUntilDeletion := deleteAt.Sub(nowFunc())
+	if timeUntilDeletion > 0 {
+		return reconcile.Result{RequeueAfter: timeUntilDeletion}, nil
+	} else {
+		err := gardenClient.Client().Delete(ctx, event)
+		if err != nil {
+			log.Debug("Failed to delete event due to %w", err)
+		}
+		return reconcile.Result{}, err
+	}
+}
+
+func longLivedEvent(event *corev1.Event) bool {
+	if event.InvolvedObject.Kind == "Shoot" {
+		if longLivedEventReasons.Has(event.Reason) {
```

Comment:
As questioned above: Should we consider all events for a `Shoot` as "long-lived" without the above whitelist?